src/helpers/rwcDuplicates.js

```diff
@@ -0,0 +1,24 @@
+import rwc from 'random-weighted-choice'
+
+/**
+ * Choose randomly a card from the cards array, based on the weights, when cards are
+ * uniquely identified by an id and an idx property
+ * Note: Use only when the cards contain an idx property
```

Comment:
Maybe you could ensure they all have an `idx`, otherwise fallback to a regular `rwd`?

```js
if (!cards.every(hasIdx)) return rwd(cards)
```

Reply:
It's probably better, but it still leaves the case of some cards having an idx and others not. This would cause their idx to be set to 'null'